### PR TITLE
VITIS-11750 Rename exe from xbutil to xrt-smi

### PR DESCRIPTION
**Pull request by rchane**
#### Problem solved by the commit
https://jira.xilinx.com/browse/VITIS-11750
Rename xbutil to xrt-smi for npu management and monitoring. SMI is an industry-recognized acronym, and "xrt-smi" was settled upon after review.
#### Bug / issue (if any) fixed, which PR introduced the bug, how it was discovered
N/A
#### How problem was solved, alternative solutions (if any) and why they were rejected
Originally, the goal was to create an xrt-smi wrapper on top of xbutil. However, that was updated to changing the exe itself to "xrt-smi" and providing an extra script to allow "xbutil" use on Linux (mainly for Alveo/legacy users) for now.

Enabled xrt-smi while keeping an "xbutil" backdoor for commands as described in the ticket. On Linux, one can use either "xrt-smi" or "xbutil" before commands. Using "xbutil" will print a warning message that it has been renamed to "xrt-smi." There are now 3 scripts above the unwrapped xrt-smi exe: xbutil, xrt-smi, and xrt-smi.bat. See the CMake in the files for more info.
#### Risks (if any) associated the changes in the commit
Documentation change will be needed. Should not break old tests as "xbutil" is still allowed for now via the xbutil script.
#### What has been tested and how, request additional testing if necessary
`xrt-smi/xrt-smi.bat examine/configure/validate` on MCDM/Strix works. `xrt-smi examine/configure/validate` and `xbutil examine/configure/validate` on Linux works. I would appreciate additional testing to make sure that no pipelines break because of this change. There may also be more cleanup work needed in output/variables to scrub out "xbutil," but the most important thing is that output on RyzenAI only mentions "xrt-smi."

When building XRT with ./build/build.sh on Linux, you will see:
```
Test project /proj/xsjhdstaff4/rchane/XRT/build/Debug
      Start  1: xrt-smi
 1/19 Test  #1: xrt-smi ...........................   Passed    0.16 sec
      Start  2: xbmgmt2
 2/19 Test  #2: xbmgmt2 ...........................   Passed    0.02 sec
[...More Output...]
```
"xbutil" is still allowed on Linux via script, but there will be a warning:
```
rchane@xsjrchane50:/proj/rdi/staff/rchane/XRT$ xbutil examine
----------------------------------------------------------------------
                              WARNING:
                xbutil has been renamed to xrt-smi
        Please migrate to using xrt-smi instead of xbutil.

    Commands, options, arguments and their descriptions can also be 
                    reported via the --help option.
----------------------------------------------------------------------
System Configuration
  OS Name              : Linux
  Release              : 6.4.0-rc7
  Version              : #2 SMP PREEMPT_DYNAMIC Thu Jun 22 14:12:00 PDT 2023
  Machine              : x86_64
  CPU Cores            : 16
  Memory               : 46830 MB
[...More Output...]
```
Linux cmd-line autocomplete works with "xrt-smi."
```
rchane@xsjrchane50:/proj/rdi/staff/rchane/XRT$ source build/Debug/opt/xilinx/xrt/setup.sh
Autocomplete enabled for the xrt-smi command
Autocomplete enabled for the xbmgmt command
XILINX_XRT        : /proj/xsjhdstaff4/rchane/XRT/build/Debug/opt/xilinx/xrt
[...More Output...]
```
Help menus on both Linux and Windows will now say "xrt-smi" instead of "xbutil," like:
```
rchane@xsjrchane50:/proj/rdi/staff/rchane/XRT$ xrt-smi examine --help

COMMAND: examine

DESCRIPTION: This command will 'examine' the state of the system/device and will generate a report of interest
               in a text or JSON format.

USAGE: xrt-smi examine [--help] [-d arg] [-f arg] [-o arg] [-r arg]
[...More Output...]
```
Some sample output on Windows:
```
C:\Users\rchane\Desktop\xilinx\xrt>xrt-smi examine --help

COMMAND: examine

DESCRIPTION: This command will 'examine' the state of the system/device and will generate a report of interest
               in a text or JSON format.

USAGE: xrt-smi examine [--help] [-d arg] [-f arg] [-o arg] [-r arg]

OPTIONS:
  -d, --device       - The Bus:Device.Function (e.g., 0000:d8:00.0) device of interest
  -f, --format       - Report output format. Valid values are:
                         JSON        - Latest JSON schema
                         JSON-2020.2 - JSON 2020.2 schema
  -o, --output       - Direct the output to the given file
  --help             - Help to use this sub-command
  -r, --report       - The type of report to be produced. Reports currently available are:
                         aie            - AIE metadata in xclbin
                         aie-partitions - AIE partition information
                         aiemem         - AIE memory tile information
                         aieshim        - AIE shim tile status
                         all            - All known reports are produced
                         electrical     - Electrical and power sensors present on the device
                         host           - Host information
                         platform       - Platforms flashed on the device


GLOBAL OPTIONS:
  --verbose          - Turn on verbosity
  --batch            - Enable batch mode (disables escape characters)
  --force            - When possible, force an operation

C:\Users\rchane\Desktop\xilinx\xrt>xrt-smi validate -r all
Validate Device           : [00c5:00:01.1]
    Platform              : NPU
    Performance Mode      : Default
-------------------------------------------------------------------------------
Test 1 [00c5:00:01.1]     : latency
    Details               : Kernel name is 'DPU_PDI_0'
                            Instruction size: '20' bytes
                            No. of iterations: '10000'
                            Average latency: '118.4' us
    Test Status           : [PASSED]
-------------------------------------------------------------------------------
Test 2 [00c5:00:01.1]     : throughput
    Details               : Kernel name is 'DPU_PDI_0'
                            Instruction size: '20' bytes
                            No. of iterations: '2500'
                            Average throughput: '6115.0' ops/s
    Test Status           : [PASSED]
-------------------------------------------------------------------------------
Test 3 [00c5:00:01.1]     : df-bw
    Details               : Kernel name is 'DPU_PDI_0'
    Details               : Buffer size: '1'GB
                            No. of iterations: '600'
                            Total duration: '21.0's
                            Average bandwidth per shim DMA: '57.1' GB/s
    Test Status           : [PASSED]
-------------------------------------------------------------------------------
Test 4 [00c5:00:01.1]     : tct-one-col
    Details               : Kernel name is 'DPU_PDI_0'
    Details               : Buffer size: '4'bytes
                            No. of iterations: '10000'
                            Average time for TCT: '2.3' us
                            Average TCT throughput: '433303.8' TCT/s
    Test Status           : [PASSED]
-------------------------------------------------------------------------------
Test 5 [00c5:00:01.1]     : tct-all-col
    Details               : Kernel name is 'DPU_PDI_0'
    Details               : Buffer size: '4' bytes
                            No. of iterations: '20000'
                            Average time for TCT: '1.2' us
                            Average TCT throughput: '865797.1' TCT/s
    Test Status           : [PASSED]
-------------------------------------------------------------------------------
Test 6 [00c5:00:01.1]     : gemm
    Details               : Kernel name is 'DPU_1x4'
    Details               : Total Duration: '126.5' ns
                            Average cycle count: '229.0'
                            IPU H-Clock: '1810' MHz
                            TOPS: '49.7'
    Test Status           : [PASSED]
-------------------------------------------------------------------------------
Validation completed. Please run the command '--verbose' option for more details

C:\Users\rchane\Desktop\xilinx\xrt>xrt-smi configure --performance performance

Performance mode is set to performance
```
#### Documentation impact (if any)
xbutil is now xrt-smi. Lots of documentation to replace the name, though functionality is still the same.

### CODE REVIEW

File: src/runtime_src/core/tools/xbutil2/npu-smi
Comment (AShivangi) [inline src/runtime_src/core/tools/xbutil2/npu-smi:35]: We don't need this anymore, but is there a way we can do something like this for "xbutil"? So if someone uses xbutil, we should inform them that this is a deprecated toolname, please use "npu-smi

**Comment by AShivangi** [comment]
Overall it looks good, but waiting on the final decision for now.

**Review by AShivangi — APPROVED** [review]

**Comment by AShivangi** [comment]
@stsoe, can you please look at this PR?

**Review by stsoe — COMMENTED** [review]
I am a little confused here?  Aren't we renaming `xbutil` to `xrt-smi`?  If so I imagine the the executable should be renamed to `xrt-smi` and that we provide xbutil scripts for backwards compatibility?

File: src/runtime_src/core/tools/xbutil2/xrt-smi
Comment (AShivangi) [inline src/runtime_src/core/tools/xbutil2/xrt-smi:45]: no one uses legacy anymore. We don't need this while loop

**Review by AShivangi — CHANGES_REQUESTED** [review]
Thanks for the verbose description!

Let's cleanup the new xrt-smi script and we should be good!

**Review by stsoe — APPROVED** [review]
Thank you for good detailed PR description.  This looks good.

**Review by chvamshi-xilinx — APPROVED** [review]
Looks good.

**Pull request merged**
